### ISSUE TITLE
Bitnami Wordpress ACI: Error executing 'postInstallation' operation not supported on socket, symlink '/opt/bitnami/wordpress/wordpress-htaccess.conf'

### ISSUE THREAD

**Issue by Air-Git** [bitnami/bitnami-docker-wordpress#172]
I have launched a new Azure Container Instance of Bitnami Wordpress, following the instructions here: https://docs.bitnami.com/azure/get-started-aci/.
Everything is created OK, but when the Wordpress container starts, it gives an error: 

```
-------------------------------------------------
nami    INFO  Initializing apache
apache  INFO  ==> Reconfiguring PID file location...
apache  INFO  ==> Configuring dummy certificates...
nami    INFO  apache successfully initialized
nami    INFO  Initializing php
nami    INFO  php successfully initialized
nami    INFO  Initializing mysql-client
nami    INFO  mysql-client successfully initialized
nami    INFO  Initializing libphp
nami    INFO  libphp successfully initialized
nami    INFO  Initializing wordpress
Error executing 'postInstallation': ENOTSUP: operation not supported on socket, symlink '/opt/bitnami/wordpress/wordpress-htaccess.conf' -> '/opt/bitnami/apache/conf/vhosts/htaccess/wordpress-htaccess.conf'
----------------------------------------------------------------------
```

The container remains in a Waiting state. Any ideas what is wrong?

**Comment by juan131** [comment]
Hi @Air-Git 

Are you using the Bitnami ACI Template available at the repository below to deploy WordPress on Azure Container Instances?

https://github.com/bitnami-labs/aci-templates/tree/master/wordpress

**Comment by miguelaeh** [comment]
Hi @Air-Git,
I have replicate your issue and you are right, this is a bug.
Thank you for reporting it. I fixed it and it should be released again in a few hours.

**Comment by Air-Git** [comment]
Brilliant, thanks for fixing it.
The container approach makes a lot of sense for this, and the Azure Template is a great way to implement it. It makes it less effort, but without hiding the details like a black box.

**Comment by Air-Git** [comment]
Deleted everything and ran the template again. Same error as before. The Wordpress container has terminated and restarted 4 times..

**Comment by miguelaeh** [comment]
Sorry, the release was not done. I have do it manually.

**Comment by Air-Git** [comment]
I have the instance running now, but there is no response from the IP address. I will open another issue.

**Comment by Air-Git** [comment]
Spoke too soon. The containers are running but the Wordpress container cannot connect to the MySQL container: with the problem as you said above. So I will close this one. But does that mean the containers will fail very time they are refreshed? Or is that sha change stored statically?
-----------------------------
mysql-c ERROR [canConnect] Connection with 'bn_wordpress' user is unsuccessful
Error executing 'postInstallation': Cannot connect to MySQL server:
MySQL command failed to run. Error: 
 ERROR 2059 (HY000): Plugin caching_sha2_password could not be loaded: /opt/bitnami/mysql//opt/bitnami/mysql/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory
----------------------------------------------

**Comment by Air-Git** [comment]
Spoke too soon. I get the error you mentioned. It takes a few minutes after starting before it gives this error.
-------------------------------------
``mysql-c ERROR [canConnect] Connection with 'bn_wordpress' user is unsuccessful
Error executing 'postInstallation': Cannot connect to MySQL server:
MySQL command failed to run. Error: 
 ERROR 2059 (HY000): Plugin caching_sha2_password could not be loaded: /opt/bitnami/mysql//opt/bitnami/mysql/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory``

**Comment by Air-Git** [comment]
I don't really get what the solution is. The container is in a Waiting state with this error. I don't see how I can make an SSH connection to it, and anyway, how would the container allow an SSH connection like this?

**Comment by Air-Git** [comment]
Does this mean that the Wordpress container cannot work, until the MySQL password problem is solved?

**Comment by miguelaeh** [comment]
Hi @Air-Git ,
> I don't see how I can make an SSH connection to it

To create an ssh conection to the container you need to use the `az` command. You have to install it and configure for your account. After that, indicate the resource group and the container to create the conection.

>Does this mean that the wordpress container cannot work, until the MySQL password problem is solved?
Yes, in the initialization stage wordpress will check if the MySQL server connection can be done. It will stay trying to connect until the MySQL server will be ready, after a timeout it will fail.

Sorry for the inconvenience, 
the error `Plugin caching_sha2_password could not be loaded` is due to an update of mysql and we have to improve it.

**Comment by Air-Git** [comment]
Ah! Thanks, I will try az.

**Comment by Air-Git** [comment]
I am getting a little further with this. For anyone who picks up this thread:
- the problem is that the SQL client in the Wordpress container does not support the default authentication method in the MySQL Server in the MySQL container. So the Wordpress database user bn_wordpress cannot authenticate to the database.
- there is no need to follow links about SSH and Az CLI. The Azure portal gives a window to connect directly to the container with a Bash shell.
- the next problem is what to change, but that's a different topic from this issue.
Bottom line is this template won't work without a solution to the authentication problem.

**Comment by miguelaeh** [comment]
Hi @Air-Git ,
>the next problem is what to change

Is it not enough with the link I provided? 
>If you get the error: plugin 'caching_sha2_password' cannot be loaded
Refer to this issue to solve it: https://community.bitnami.com/t/authentication-plugin-caching-sha2-password-cannot-be-loaded-mysql-workbbench/58223/4

**Comment by Air-Git** [comment]
The links gives some clues, but not how to solve it in the container. I can get onto the console directly from within the Azure portal. But when I run the command ALTER USER etc. I get an error. I have opened a separate ticket, as it is different from the template error that this ticket was about.

**Comment by miguelaeh** [comment]
Okay, thanks.
In that case I am going to close this now. Feel  free to open it again.

**Comment by davidcv5** [comment]
@miguelaeh could you please share what's the fix for this? it's also happening for another bitnami image.
https://github.com/bitnami/bitnami-docker-ghost
Exact same error:
```
Error executing 'postInstallation': ENOTSUP: operation not supported on socket, symlink '/opt/bitnami/ghost/logs' -> '/bitnami/ghost/content/logs'
```

**Comment by miguelaeh** [comment]
Hi @davidcv5 ,
This is not an error on the image itself, we use a symlink on the initialization logic and it seems that Azure is not supporting the use of symlinks.
To solve the previous one I just changed the symlink by moving the file in the container initialization logic.
Thank you very much for reporting the issue, we will fix it for the Ghost container too.
We will update this thread when we fix it.

**Comment by miguelaeh** [comment]
As you created another issue for the same purpose, we will use it instead of this that is closed.
Find the issue [here](https://github.com/bitnami/bitnami-docker-ghost/issues/79)
